### PR TITLE
Use flate2 crate to tar up the OUTPUT

### PR DESCRIPTION
**Pull request by bencord0**
We already depend on flate2 for tar.gz unpacking,
use if for creating archives too.

In addition, this helps us archive artifacts with lots
of OUTPUT files, instead of getting the following error.

```
lal::build: Build succeeded with verified dependencies
lal::core::output: Taring OUTPUT

lal: build error: Argument list too long (os error 7)
```

### CODE REVIEW

File: src/core/output.rs [hunk 1]
```diff
@@ -1,40 +1,24 @@
...
 
-    // append each file as an arg to the main tar process
-    for f in find_str.trim().split('\n') {
-        args.push(f.into())
-    }
+    // Don't dereference symlinks, archive them as-is.
+    // Dereferencing symlinks makes the archive larger, as we will then store two copies  of the
+    // same file. Additionally, if OUTPUT contains dangling symlinks, adding them to the archive
+    // will fail with a NotFound error.
+    archive.follow_symlinks(false);
```
Comment (clux) [inline src/core/output.rs:19]: Ok. This sounds good. (Had to re-read this a few times as it made it sound like we were doubling our tarball sizes in some cases.)

So we only allow: symlinks that point to something relative to (and within) OUTPUT.
And everything else is the same when you inspect a tarball with `tar`?

File: src/core/output.rs [hunk 2]
```diff
@@ -1,40 +1,24 @@
...
+use super::{LalResult};
 
 /// Helper for stash and build
 pub fn tar(component_dir: &Path, tarball: &Path) -> LalResult<()> {
     info!("Taring OUTPUT");
-    let mut args: Vec<String> = vec![
-        "czf".into(),
-        tarball.to_str().unwrap().into(), // path created internally - always valid unicode
-        "--transform=s,^OUTPUT/,,".into(), // remove leading OUTPUT
-    ];
```
Comment (clux) [inline src/core/output.rs:12]: Man, wish I had tests for this. Remember this was very subtle. I think we wanted compatibility with `tar` so the stuff we compressed could be piped to tar and extracted as well for debugging. Not sure if flate2 handled it correctly, and that might have been the reason for it (it should be in the history).

**Review by clux — COMMENTED** [review]
question on `tar` compatibility - otherwise it looks good in terms of reducing complexity.

File: src/core/output.rs (hunk 2 above)
Comment (bencord0) [reply src/core/output.rs:12]: > stuff we compressed could be piped to tar.

I'll add a test for this. After reshuffling, I think this makes sense to add this to the `export` tests.

File: src/core/output.rs (hunk 1 above)
Comment (bencord0) [reply src/core/output.rs:19]: Hmm, we should definitely add a check (and enforce with tests) to ensure that symlinks can only be relative to OUTPUT.

I've definitely managed to accidentally create absolute symlinks in local testing.
My current workaround was to write a `rewrite_symlinks.sh` script, but perhaps we should fail the build (or `build --release` at the very least) step.

**Comment by bencord0** [comment]
Closing this PR in favour of #59 which will have access to CI secrets

**Pull request closed without merging**